server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -464,30 +463,56 @@ describe("Routerlicious", () => {
 							);
 							expectedSignals.forEach((signal, signalIndex) => {
 								const receivedSignal = client.signalsReceived[signalIndex];
-								Object.keys(signal).forEach((property) => {
-									assert.deepEqual(
-										receivedSignal[property],
-										signal[property],
-										`User ${index + 1} signal ${
-											signalIndex + 1
-										} ${property} mismatch`,
-									);
-								});
+								assert.deepEqual(
+									receivedSignal,
+									signal,
+									`received signal does not match expected signal`,
+								);
 							});
 						});
 					}
 
+					function isSentSignalMessage(obj: unknown): obj is ISentSignalMessage {
+						return (
+							typeof obj === "object" &&
+							obj !== null &&
+							"content" in obj &&
+							(!("type" in obj) || typeof obj.type === "string") &&
+							(!("clientConnectionNumber" in obj) ||
+								typeof obj.clientConnectionNumber === "number") &&
+							(!("referenceSequenceNumber" in obj) ||
+								typeof obj.referenceSequenceNumber === "number") &&
+							(!("targetClientId" in obj) || typeof obj.targetClientId === "string")
+						);
+					}
+
+					function sendValidAndReturnExpectedSignal(
+						client: TestSignalClient,
+						content: unknown,
+					): ISignalMessage {
+						const signal = client.version === 2 ? { content } : content;
+						return sendAndReturnExpectedSignal(client, signal);
+					}
+
 					function sendAndReturnExpectedSignal(
 						client: TestSignalClient,
 						signal: unknown,
 					): ISignalMessage {
 						client.socket.send("submitSignal", client.clientId, [signal]);
 						let expectedSignalMessage: ISignalMessage;
-						if (client.version === 2 && isSentSignalMessage(signal)) {
-							expectedSignalMessage = {
-								...signal,
-								clientId: client.clientId,
-							};
+						if (client.version === 2) {
+							if (isSentSignalMessage(signal))
+								expectedSignalMessage = {
+									...signal,
+									clientId: client.clientId,
+								};
+							else {
+								// Dummy signal for v2 clients
+								expectedSignalMessage = {
+									clientId: undefined,
```

Comment:
We let `clientId` be returned as `undefined`?

Reply:
This dummy signal is just set up purposely for the test to fail after the type check

Reply:
I understand. I just didn't think that `undefined` would be an allowable value for `ISignalMessage.clientId`.
I looked up the type. It is string | null. I think that server isn't using strict enough typescript to flag this. I suggest using a dummy (invalid client id) string value. Then nothing to clean up if strictness is increased, which it should.